### PR TITLE
Backend JWT & JWKS docs (Master)

### PR DESCRIPTION
**Pull request by jhivandebenoit**
issues: [1] (https://github.com/wso2/product-microgateway/issues/3140)

### CODE REVIEW

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md [hunk 1]
```diff
@@ -163,18 +163,162 @@ The relevant elements in the JWT generation configuration are described below. I
...
-<tr class="even">
-<td><pre><code>enforcer.jwtGenerator.privateKeyPath</code></pre></td>
+<tr class="odd">
+<td><pre><code>enforcer.jwtGenerator.keypair.privateKeyPath</code></pre></td>
 <td><p>Path of the private key</p></td>
 <td><code>/home/wso2/security/keystore/mg.key</code></td>
 </tr>
+<tr class="even">
+<td><pre><code>enforcer.jwtGenerator.keypair.useForSigning</code></pre></td>
+<td><p>Whether to use key for signing</p></td>
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md:201]:
```suggestion
<td><p>Whether to use key for signing.</p></td>
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md [hunk 2]
```diff
@@ -163,18 +163,162 @@ The relevant elements in the JWT generation configuration are described below. I
...
+<td><code>false</code></td>
+</tr>
+<tr class="odd">
+<td><pre><code>enforcer.jwtGenerator.jwksRateLimitQuota</code></pre></td>
+<td><p>How many requests to the JWKS endpoint can be served in the time window</p></td>
+<td><code>1000</code></td>
+</tr>
+<tr class="even">
+<td><pre><code>enforcer.jwtGenerator.keypair</code></pre></td>
+<td><p>An object containing the paths to public certificate and private key of an RSA keypair and specifying whether to use it for signing or not</p></td>
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md:182]:
```suggestion
<td><p>An object containing the paths to public certificate and private key of an RSA keypair and specifying whether to use it for signing or not.</p></td>
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md [hunk 3]
```diff
@@ -163,18 +163,162 @@ The relevant elements in the JWT generation configuration are described below. I
...
+<td><pre><code>kid</code></pre></td>
+<td>This is an id parameter used to match a specific key(s)</td>
+</tr>
+<tr class="odd">
+<td><pre><code>alg</code></pre></td>
+<td>This defines the specific algorithm intended for use with the key.</td>
+</tr>
+<tr class="even">
+<td><pre><code>n</code></pre></td>
+<td>The modulus value of the public key</td>
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md:275]:
```suggestion
<td>The modulus value of the public key.</td>
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md [hunk 4]
```diff
@@ -163,18 +163,162 @@ The relevant elements in the JWT generation configuration are described below. I
...
+</tr>
+<tr class="odd">
+<td><pre><code>keys</code></pre></td>
+<td>An array of the public keys available from the gateway</td>
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md:279]:
```suggestion
<td>An array of the public keys available from the gateway.</td>
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md [hunk 5]
```diff
@@ -163,18 +163,162 @@ The relevant elements in the JWT generation configuration are described below. I
...
+        if result is jwt:Error {
+            return error("Failed to authenticate " + result.message());
+        }
+        
+        return result.toBalString();
+    }
+}
+```
+
+In this example we create a hello route on port 8080 and secure it with JWT. this hello function returns the JWT claims or the error that occured during validation.
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md:320]:
```suggestion
In this example we create a hello route on port 8080 and secure it with JWT. This hello function returns the JWT claims or the error that occurred during validation. 
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md [hunk 6]
```diff
@@ -163,18 +163,162 @@ The relevant elements in the JWT generation configuration are described below. I
...
+<td><pre><code>e</code></pre></td>
+<td>The exponent value of the public key.</td>
+</tr>
+<tr class="odd">
+<td><pre><code>use</code></pre></td>
+<td>This defines the use of the key, whether it is used for signing or encryption.</td>
+</tr>
+<tr class="even">
+<td><pre><code>kid</code></pre></td>
+<td>This is an id parameter used to match a specific key(s)</td>
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md:267]:
```suggestion
<td>This is an ID parameter used to match a specific key(s).</td>
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md [hunk 7]
```diff
@@ -163,18 +163,162 @@ The relevant elements in the JWT generation configuration are described below. I
...
+<table>
+<thead>
+<tr class="header">
+<th><b>Parameter</b></th>
+<th><b>Description</b></th>
+</tr>
+<tbody>
+<tr class="odd">
+<td><pre><code>kty</code></pre></td>
+<td>Key type identifies the cryptographic family this key belongs to</td>
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md:255]:
```suggestion
<td>Key type identifies the cryptographic family this key belongs to.</td>
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md [hunk 8]
```diff
@@ -163,18 +163,162 @@ The relevant elements in the JWT generation configuration are described below. I
...
 </tr>
 <tr class="odd">
-<td><pre><code>enforcer.jwtGenerator.publicCertificatePath</code></pre></td>
+<td><pre><code>enforcer.jwtGenerator.jwksRateLimitQuota</code></pre></td>
+<td><p>How many requests to the JWKS endpoint can be served in the time window</p></td>
+<td><code>1000</code></td>
+</tr>
+<tr class="even">
+<td><pre><code>enforcer.jwtGenerator.jwksRateLimitTimeWindow</code></pre></td>
+<td><p>Time window for the rate limit to reset</p></td>
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md:172]:
```suggestion
<td><p>Time window for the rate limit to reset.</p></td>
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md [hunk 9]
```diff
@@ -163,18 +163,162 @@ The relevant elements in the JWT generation configuration are described below. I
 <td><code>org.wso2.carbon.apimgt.impl.token.ExtendedDefaultClaimsRetriever</code></td>
 </tr>
 <tr class="odd">
-<td><pre><code>enforcer.jwtGenerator.publicCertificatePath</code></pre></td>
+<td><pre><code>enforcer.jwtGenerator.jwksRateLimitQuota</code></pre></td>
+<td><p>How many requests to the JWKS endpoint can be served in the time window</p></td>
```
Comment (Mariangela) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/passing-enduser-attributes-to-the-backend-via-choreo-connect.md:167]:
```suggestion
<td><p>How many requests to the JWKS endpoint can be served in the time window.</p></td>
```

**Review by Mariangela — APPROVED** [review]

**Review by Mariangela — APPROVED** [review]

**Comment by Mariangela** [comment]
@jhivandebenoit , please check and send these changes to the 4.2.0 branch if you have not done so already.

**Comment by jhivandebenoit** [comment]
Changes have been send and merged

**Pull request merged**